volatility3/framework/symbols/windows/extensions/consoles.py

```diff
@@ -0,0 +1,439 @@
+# This file is Copyright 2024 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+from volatility3.framework import objects
+from volatility3.framework import constants
+
+
+class ROW(objects.StructType):
+    """A Row Structure."""
+
+    def _valid_dbcs(self, c, n):
+        # TODO this need more research and testing
+        # https://github.com/search?q=repo%3Amicrosoft%2Fterminal+DbcsAttr&type=code
+        valid = n == 0 and c in (
+            0x0,
+            0x1,
+            0x2,
+            0x8,
+            0x10,
+            0x18,
+            0x20,
+            0x28,
+            0x30,
+            0x48,
+            0x50,
+            0x58,
+            0x60,
+            0x68,
+            0x70,
+            0x78,
+            0x80,
+            0x88,
+            0xA8,
+            0xB8,
+            0xC0,
+            0xC8,
+            0x98,
+            0xD8,
+            0xE0,
+            0xE8,
+            0xF8,
+            0xF0,
+            0xA0,
+        )
+        if n == 0 and not valid:
+            print("Bad Dbcs Attribute {}".format(hex(c)))
+        return valid
+
+    def get_text(self, truncate=True):
```

Comment:
Presumably this returns a `str`?  Please could we typing information in the signature?

Reply:
done